src/components/ListItem.js

```diff
@@ -34,14 +35,39 @@ const ListItem = ({ listItem, itemId }) => {
   const isChecked =
     fromMilliSecToHours(getUTCNowInMilliSec() - listItem.recentPurchase) < 24;
 
+  // This function specifies what color each item should have based on how soon the item is to be bought.
+  // also return arial-label each item should have
+  const getStatusAndAriaLabel = ({ howSoon, recentPurchase }) => {
+    if (isOutOfDate(howSoon, recentPurchase)) {
+      return ['inactive', 'inactive item'];
+    }
+
+    return howSoon <= 7
+      ? ['soon', 'Next purchase within 7 days']
+      : howSoon > 7 && howSoon < 30
+      ? ['kind-of-soon', 'Next purchase within 30 days']
+      : ['not-soon', 'Next purchase within more than 30 days'];
+  };
+
+  const [status, ariaLabel] = getStatusAndAriaLabel(listItem);
+
   return (
-    <li key={listItem.name} className="list-item">
+    <li
+      key={listItem.name}
+      className={`list-item ${status}-item`}
+      aria-label={ariaLabel}
```

Comment:
This aria-label was ignored by screen reader when I tried it.
My thoughts of making the list accessible is to give this arialabel to the item name in line 72 and also give it tab-index = 0 so it's not ignored by screen readers and move it up before the input because it makes more sense to say the item name first (you can adjust the order as you like in CSS using `order`)
also add item name to arialabel so it's more descriptive (apple Next purchase within 7 days) and if it's disabled you can add at the end (disabled  for 24 hours)
```html
<li
      key={listItem.name}
      className={`list-item ${status}-item`}
    >
      <div className="item-name" tabindex="0" aria-label={ariaLabel}>{listItem.name}</div>
      <input
        aria-label="check to mark as purchased"
        type="checkbox"
        className="check-item"
        onChange={() => checkItem()}
        disabled={isChecked}
        checked={isChecked}
      />
</li>
```

Reply:
I'll try to work on it tonghit. :)